libraries/botbuilder-core/botbuilder/core/middleware_set.py

```diff
@@ -32,17 +32,18 @@ def __init__(self):
         super(MiddlewareSet, self).__init__()
         self._middleware = []
 
-    def use(self, middleware: Middleware):
+    def use(self, *middleware: Middleware):
         """
         Registers middleware plugin(s) with the bot or set.
         :param middleware :
         :return:
         """
-        if hasattr(middleware, 'on_process_request') and callable(middleware.on_process_request):
-            self._middleware.append(middleware)
-            return self
-        else:
-            raise TypeError('MiddlewareSet.use(): invalid middleware being added.')
+        for (idx, m) in enumerate(middleware):
+            if hasattr(m, 'on_process_request') and callable(m.on_process_request):
+                self._middleware.append(m)
+                return self
+            else:
+                raise TypeError('MiddlewareSet.use(): invalid middleware at index "%s" being added.' % idx)
 
     async def receive_activity(self, context: BotContext):
```

Comment:
In receive_activity_internal log: try/except & log

Reply:
done!